src/mds/MDCache.cc

```diff
@@ -271,6 +271,9 @@ void MDCache::add_inode(CInode *in)
         g_conf->mds_cache_size * g_conf->mds_health_cache_threshold) {
     exceeded_size_limit = true;
   }
+
+  /* This is the place where inodes are thrown in the cache, so check if the inode should be somewhere else: */
+  in->maybe_export_pin();
```

Comment:
It's not true. see CDir::_load_dentry. Maybe we should call maybe_export_pin in CDir::link_primary_inode